ccube.nim

```diff
@@ -66,142 +66,101 @@ Examples:
-      noLoopSwitch = "--noloop:"
-      fileSwitch = "--file:"
-      numberParam = "-n:"
-      filePath:string
-      number:string
-      noLoopSwitchBool:string
-    for i in 1..paramCount():
-      paramSequence.add(paramStr(i))
-    for param in paramSequence:
-      if noLoopSwitch in param:
-        try:
-          noLoopSwitchBool = param.split(':')[1]
-          if noLoopSwitchBool in validBools:
-            discard
-          else:
-            stderr.write(&"Error:\"{noLoopSwitchBool}\",Invalid bool used")
-            quit()
-        except IndexDefect:
-          stderr.write("Error:Bad use of --noloop switch,valid bool expected")
-          quit()
-      
-      if numberParam in param:
-        try:
-          number = param.split(':')[1]
-          for forbiddenChar in forbiddenCharArray:
-            if forbiddenChar.toUpper in number or forbiddenChar.toLower in number:
-              stderr.write(fmt"Error:Expected a valid integer but got '{number}',[Value Error]")
-              quit()
-        except IndexDefect:
-          stderr.write("Error:Bad use of --noloop switch,valid integer expected")
-          quit()
-      
-      if fileSwitch in param:
+
+type
+  ProgramArgs = object
+    noLoopSwitch: bool
+    number: string
+    filePath: string
+
+proc parseParams(): ProgramArgs =
+  if paramCount() == 0:
+    showUsage()
+  
+  var p = initOptParser()
+  for kind, key, val in p.getopt():
+    case kind
+    of cmdEnd: break
+    of cmdShortOption, cmdLongOption:
+      case key
+      of "noloop":
         try:
-          filePath = param.split(':')[1]      
-        except IndexDefect:
-          stderr.write("Error:Bad use of --file switch,valid file path expected") 
+          result.noLoopSwitch = parseBool(val)
+        except ValueError:
+          stderr.write(fmt"Error: '{val}' is not a valid boolean value.")
           quit()
-    
-    if len(filePath) > 0:
-      discard
-    else:
-      filePath = fmt"{getCurrentDir()}{DirSep}iterationData.txt"
-    
-    if len(number) > 0:
-      discard
-    else:
-      number = $rand(1..1000)
-    
-    if len(noLoopSwitchBool) > 0:
-      discard
-    else:
-      noLoopSwitchBool = "false"
-    
-    result = (noLoopSwitchBool,number,filePath)
-  
-  else:
-    showUsage()
+      of "n":
+        for c in val:
+          if c notin Digits:
+            stderr.write(fmt"Error: '{val}' is not a valid number.")
+            quit()
+        result.number = val
+      of "file":
+        result.filePath = val
+    else: discard
 
-proc isEven(number:BigInt):bool =
-  if number mod 2 == 0:
-    result = true
-  else:
-    result = false
+  if result.filePath.len == 0:
+    result.filePath = getCurrentDir() / "iterationData.txt"
 
-proc readJson():(string,string,string) =
-  var isFile = fileExists("ccube_conf.json")
+  if result.number.len == 0:
+    result.number = $rand(1..1000)
+
+proc isEven(number: BigInt): bool =
+  result = number mod 2 == 0
```

Comment:
In Nim you can assign boolean expressions to variables directly, so you don't need an if statement in this proc. If you want you can even inline it after the PR.